subgraphs/aave-v2/src/mappings/utilFunctions.ts

```diff
@@ -0,0 +1,559 @@
+  const staFees = staAmountUSD.times(market.stableBorrowRate);
+
+  // Add these values together, save to market and add protocol total
+  market.totalRevenueUSD = staFees.plus(varFees).truncate(3);
+  protocol.totalRevenueUSD = protoMinusMarketFees.plus(market.totalRevenueUSD);
+  market.protocolSideRevenueUSD = market.totalRevenueUSD.times(bigIntToBigDecimal(market.reserveFactor, 4)).truncate(3);
+  protocol.protocolSideRevenueUSD = protoMinusMarketProtoRevenue.plus(market.protocolSideRevenueUSD);
+  market.supplySideRevenueUSD = market.totalRevenueUSD.times(BIGDECIMAL_ONE.minus(bigIntToBigDecimal(market.reserveFactor, 4))).truncate(3);
+  protocol.supplySideRevenueUSD = protoMinusMarketSupplyRevenue.plus(market.supplySideRevenueUSD);
+
+  // CALCULATE totalBorrowUSD FIELDS ON MARKET AND PROTOCOL ENTITIES
+  // The sum in USD of s and v tokens on market is totalBorrowUSD
+  // Subtract the previously saved amount of the market TotalBorrowUSD value from the protocol totalBorrowUSD
+  // This gets the amount in USD out in borrows on the protocol not including this market
+  const tempProtocolBorrowTotal = protocol.totalBorrowUSD.minus(market.totalBorrowUSD);
+  // Sum the amount in USD of the stable borrows and variable borrows currently in use
+  market.totalBorrowUSD = staAmountUSD.plus(varAmountUSD);
+  // Add this markets new amount out in borrows to the protocol value
+  protocol.totalBorrowUSD = tempProtocolBorrowTotal.plus(market.totalBorrowUSD);
+
+  market.save();
+  protocol.save();
+}
+
+export function updateTVL(token: Token, market: Market, protocol: LendingProtocol, amountInTokens: BigInt, toSubtract: bool): void {
+  // Update the total value locked in a market and the protocol overall after transactions
+  let newMarketTVL = market.inputTokenBalances[0];
+  if (toSubtract) {
+    newMarketTVL = newMarketTVL.minus(amountInTokens);
+  } else {
+    newMarketTVL = newMarketTVL.plus(amountInTokens);
+  }
+  // Subtract the PREVIOUSLY ADDED totalValueLockedUSD of the market from the protocol TVL
+  // Subtracting the most recently added TVL of the market ensures that the correct
+  // proportion of this market is deducted before adding the new market TVL to the protocol
+  // Otherwise, the difference in asset USD/ETH price  since saving would deduct the incorrect proportion from the protocol TVL
+  protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.minus(market.totalValueLockedUSD);
+  market.totalValueLockedUSD = amountInUSD(market.inputTokenPricesUSD[0], token.decimals, newMarketTVL, market);
+  market.totalDepositUSD = market.totalValueLockedUSD;
+  market.save();
+  protocol.totalValueLockedUSD = protocol.totalValueLockedUSD.plus(market.totalValueLockedUSD);
+  protocol.totalDepositUSD = protocol.totalValueLockedUSD;
+  protocol.save();
+}
+
+export function emissionsPerDay(rewardRatePerSecond: BigInt): BigInt {
```

Comment:
Can you make `BigInt.fromString('10').pow(18)` a constant and put it in `constants.ts` and added to the reference subgraph? I think others will need it too.

Also can you put `emissionsPerDay`, `getDaysSinceEpoch`, `rayToWad`, `wadToRay` to the reference subgraph as well?

Reply:
Added these to the appropriate 'common' files in the Aave project directory and then to the current '_reference' common files.

Reply:
Added these to the appropriate 'common' files in the Aave project directory and then to the current '_reference' common files.